### PR TITLE
fix: Fixed an issue where events were overlaid when the schedule included a travel time

### PR DESCRIPTION
**Pull request by jungeun-cho**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

#### AS-IS
When a schedule is included in travel time, a conflict occurs.
스크린샷 2020-01-16 20 13 22
#### TO-BE
스크린샷 2020-01-16 20 17 48

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/js/common/array.js [hunk 1]
```diff
@@ -221,45 +224,155 @@ function stringDESCIgnoreCase(_a, _b) {
...
+ * @param {Schedule|ScheduleViewModel} a The object schedule instance.
+ * @param {Schedule|ScheduleViewModel} b The object schedule instance.
+ * @returns {number} Result of comparison.
+ */
+function _getStartCompare(a, b) {
+    var aTravelTime = getMaxTravelTime(a);
+    var bTravelTime = getMaxTravelTime(b);
+    var aGoingDuration = datetime.millisecondsFrom('minutes', aTravelTime.maxGoingDuration);
+    var bGoingDuration = datetime.millisecondsFrom('minutes', bTravelTime.maxGoingDuration);
+    var renderedStartA = new TZDate(a.getStarts().getTime() - aGoingDuration);
```
Comment (dongsik-yoo) [inline src/js/common/array.js]: Why do you use TZDate? I think just plus and minus are enough.

File: src/js/common/array.js [hunk 2]
```diff
@@ -221,45 +224,155 @@ function stringDESCIgnoreCase(_a, _b) {
...
+
+/**
+ * Filter duplicate elements in arrays except the first
+ *
+ * @param {Schedule|ScheduleViewModel[]} array The List of ScheduleViewModel(or Schedule)
+ * @param {function} condition Callback function to check filtering condition
+ * @returns {Schedule|ScheduleViewModel[]} Result of filtered
+ */
+function _duplicates(array, condition) {
+    return array.slice(1, array.length).filter(condition);
```
Comment (dongsik-yoo) [inline src/js/common/array.js]: The second parameter of `slice` is not necessary..

File: src/js/common/array.js [hunk 3]
```diff
@@ -221,45 +224,155 @@ function stringDESCIgnoreCase(_a, _b) {
...
+    return 0;
+}
+
+/**
+ * Sort elements in duplicate arrays sequentially
+ *
+ * @param {Schedule|ScheduleViewModel} array The scheduleViewModel instance.
+ * @returns {Schedule|ScheduleViewModel[]} Sorted array
+ */
+function sortByDuplicate(array) {
```
Comment (dongsik-yoo) [inline src/js/common/array.js]: This method should not be in array module. I think it should be in `src/js/controller/viewMixin/week.js` for managing schedule collection.

File: src/js/common/array.js (hunk 1 above)
Comment (jungeun-cho) [reply src/js/common/array.js]: When passing as a parameter to `datetime.compare`, the function requires a `Date` object.

```js
compare: function(d1, d2) {
   var _d1 = d1.getTime(),
       _d2 = d2.getTime();
...
```

Would it be better to make it a Date object when calling the compare function?

```js
return datetime.compare(new TZDate(renderedStartA), new TZDate(renderedStartB));
```

File: src/js/common/array.js (hunk 3 above)
Comment (jungeun-cho) [reply src/js/common/array.js]: I think so too.

File: src/js/common/array.js (hunk 2 above)
Comment (jungeun-cho) [reply src/js/common/array.js]: Right.
You do not have to specify the second parameter.

> Zero-based index before which to end extraction. `slice` extracts up to but not including *end*.

**Pull request merged**